### PR TITLE
Clean up `DeviceInternals`

### PR DESCRIPTION
**Pull request by hvy**
and some cosmetics. Addresses
- https://github.com/chainer/chainer/pull/6820#discussion_r273831592
- https://github.com/chainer/chainer/pull/6820#discussion_r273832130

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/cuda_device.h [hunk 1]
```diff
@@ -52,14 +52,16 @@ class MemoryKeeper {
...
+// These internals are exposed through `GetDeviceInternals` for CUDA internal usages.
 class DeviceInternals {
 public:
     DeviceInternals(const DeviceInternals&) = delete;
     DeviceInternals(DeviceInternals&&) = delete;
     DeviceInternals& operator=(const DeviceInternals&) = delete;
     DeviceInternals& operator=(DeviceInternals&&) = delete;
 
-    explicit DeviceInternals(int index) : index_{index}, cublas_handle_{index}, cudnn_handle_{index} {}
+    explicit DeviceInternals(int index) : cublas_handle_{index}, cudnn_handle_{index} {}
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device.h]: Let's be explicit
```suggestion
    explicit DeviceInternals(int device_index) : cublas_handle_{device_index}, cudnn_handle_{device_index} {}
```

Comment (hvy) [reply chainerx_cc/chainerx/cuda/cuda_device.h]: Thanks.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**